prbench/src/prbench/envs/geom3d/base_env.py

```diff
@@ -477,8 +491,50 @@ def _create_state_dict(
                 feats["x"] = pose.position[0]
                 feats["y"] = pose.position[1]
                 feats["z"] = pose.position[2]
+            # Handle triangles.
+            elif object_type == Geom3DTriangleType:
+                body_id = self._object_name_to_pybullet_id(object_name)
+                pose = get_pose(body_id, self.physics_client_id)
+                pose_feat_names = [
+                    "pose_x",
+                    "pose_y",
+                    "pose_z",
+                    "pose_qx",
+                    "pose_qy",
+                    "pose_qz",
+                    "pose_qw",
+                ]
+                pose_feats = list(pose.position) + list(pose.orientation)
+                for feat_name, feat in zip(pose_feat_names, pose_feats, strict=True):
+                    feats[feat_name] = feat
+                # Add grasp active.
+                if self._grasped_object == object_name:
+                    feats["grasp_active"] = 1
+                else:
+                    feats["grasp_active"] = 0
+                # Triangle-specific features (fallback to 0).
+                # Here we use _get_triangle_features implemented within the envs.
+                try:
+                    a, b, depth, ttype = self._get_triangle_features(object_name)
+                except AttributeError:
```

Comment:
I don't think we should catch this error, it would indicate that something is unexpectedly wrong